### DISCUSSION TITLE
MiKTeX installation on Ubuntu 20.10

### DISCUSSION THREAD

**Discussion by groupsvkg** [MiKTeX/miktex#680] [Q&A]
How can I install MikTex on Ubuntu 20.10?

**Comment by edocevoli** [comment]
For now, only LTS releases are officially supported. A MiKTeX package for Ubuntu 21.04 will be released in 2022/Q2.

That said, it is not possible to install MiKTeX easily on Ubuntu 20.10.

**Reply by Traap** [reply]
I used the following script to setup MiKTeX on WSL2 running Ubuntu 20.04

**Script variables:**
```bash
miktexGpgKey='D6BC243565B2087BC3F897C9277A7293F59E4889'
miktexSource='ubuntu focal universe'
```

**Installation script snippet.** 
```bash
# {{{ MiKTeX

installMikTeX() {
  if [[ $miktexFlag == 1 ]]; then

    # Register GPG key for Ubuntu and Linux Mint.
    sudo apt-key adv \
         --keyserver hkp://keyserver.ubuntu.com:80 \
         --recv-keys $miktexGpgKey

    # Installation source is defined in config file. 
    echo "deb http://miktex.org/download/${miktexSource}" \
      | sudo tee /etc/apt/sources.list.d/miktex.list

    # Update database
    sudo apt-get update

    # MiKTeX
    sudo apt-get -y install \
                    miktex \
                    latexmk


    # Finish MikTeX shared installation setup.
    sudo miktexsetup --shared=yes finish
    sudo initexmf --admin --set-config-value [MPM]AutoInstall=1

    # The MiXTeX team told me to update the package database twice.  See:
    # https://github.com/MiKTeX/miktex/issues/724
    sudo mpm --admin --update
    mpm --update
    sudo mpm --admin --update
    mpm --update

  fi
}

# -------------------------------------------------------------------------- }}}
```

**The complete script is here:**  
https://github.com/Traap/debian-bootstrap

Maybe it will help.

**Reply by Traap** [reply]
As of 2021-03-26, I experience the challenge seen below:

```bash
mpm --update
```
Fails with:
```log
Sorry, but "MiKTeX Package Manager did not succeed.
log4cxx: No appender could be found for logger (mpmcli).
log4cxx: Please initialize the log4cxx system properlty.
```

I'll post back any findings.  Pointers or suggestions are welcomed.

**Reply by ArnyminerZ** [reply]
Hi, did you manage to install correctly on 22.10?

**Reply by Traap** [reply]
> Hi, did you manage to install correctly on 22.10?

I maintain https://github.com/traap/bootstrap-ubuntu to address Ubuntu applications.   Checkout the install.sh script.  There are two functions installMikTeX and installTexLive.  I set the appropriate flag in the config file.  Lately I have been using installTexLive to meet my needs.

**Reply by ArnyminerZ** [reply]
Okay, I will try, thank you

**Comment by koppor** [comment]
For the googlers "just" wanting to get MiKTeX running on their Linux machine or on GitHub actions, I have an example GitHub actions workflow at https://github.com/koppor/mwe-miktex/blob/630594c186d7d81573defbd9adfeda6743c971bf/.github/workflows/miktex.yml#L33 - there is also an updated Dockerfile at https://github.com/koppor/mwe-miktex/blob/main/Dockerfile
